homeassistant/components/ecowitt/diagnostics.py

```diff
@@ -0,0 +1,39 @@
+"""Provides diagnostics for EcoWitt."""
+from __future__ import annotations
+
+from typing import Any
+
+from aioecowitt import EcoWittListener
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntry
+
+from .const import DOMAIN
+
+
+async def async_get_device_diagnostics(
+    hass: HomeAssistant, entry: ConfigEntry, device: DeviceEntry
+) -> dict[str, Any]:
+    """Return diagnostics for a device entry."""
+    ecowitt: EcoWittListener = hass.data[DOMAIN][entry.entry_id]
+    station_id = min(device.identifiers)[1]
```

Comment:
you can't assume that only the ecowit identifiers are attached to a device. Filter it out by looking at first entry being `DOMAIN`.
```suggestion
    station_id = next(item[1] for item in device.identifiers if item[0] == DOMAIN)
```